tlatools/org.lamport.tlatools/test/pcal/parser/PcalParserTranslator.java

```diff
@@ -0,0 +1,184 @@
+package pcal.parser;
+
+import java.util.Arrays;
+import java.util.Vector;
+
+import util.AstNode;
+import util.AstNode.Kind;
+
+import pcal.AST;
+import pcal.PcalCharReader;
+import pcal.TLAtoPCalMapping;
+import pcal.TLAExpr;
+import pcal.TLAToken;
+import pcal.ParseAlgorithm;
+import pcal.PcalParams;
+import pcal.exception.ParseAlgorithmException;
+
+import org.junit.Assert;
+
+/**
+ * Utilities for parsing PlusCal and converting it to a form compatible with
+ * the unified TLA+ syntax parser test corpus.
+ */
+public class PcalParserTranslator {
+	
+	/**
+	 * Simple data class holding info about the start of a PlusCal block.
+	 */
+	private static class PcalStart {
+		/**
+		 * The line on which the PlusCal block starts.
+		 */
+		int line = 0;
+		
+		/**
+		 * The column in which the PlusCal block starts, directly after the
+		 * --algorithm or --fair algorithm tokens.
+		 */
+		int column = 0;
+		
+		/**
+		 * Whether the PlusCal block is a fair algorithm.
+		 */
+		boolean isFair = false;
+		
+		/**
+		 * Finds the start of the PlusCal block.
+		 * 
+		 * @param input Text containing PlusCal block, split into lines.
+		 */
+		public PcalStart(Vector<String> input) {
```

Comment:
Nit: I think here you can use the more general `List` type instead of `Vector`.

(Long term we want to get away from `Vector`, since its operations are all `synchronized`, creating a little bit of needless overhead when the list is not actually shared between threads.)